sys/arm64/arm64/machdep.c

```diff
@@ -415,6 +429,47 @@ set_dbregs32(struct thread *td, struct dbreg32 *regs)
 }
 #endif
 
+#if __has_feature(capabilities)
+int
+fill_capregs(struct thread *td, struct capreg *regs)
+{
+	struct trapframe *frame;
+	int i;
+
+	frame = td->td_frame;
+	regs->sp = frame->tf_sp;
+	regs->lr = frame->tf_lr;
+	regs->elr = frame->tf_elr;
+	regs->ddc = frame->tf_ddc;
+
+	for (i = 0; i < nitems(frame->tf_x); i++) {
+		regs->x[i] = frame->tf_x[i];
+		if (cheri_gettag((void * __capability)frame->tf_x[i]))
+			regs->tagmask |= (uint64_t)1 << i;
+	}
+	if (cheri_gettag((void * __capability)frame->tf_lr))
+		regs->tagmask |= (uint64_t)1 << i;
+	i++;
+	if (cheri_gettag((void * __capability)frame->tf_sp))
+		regs->tagmask |= (uint64_t)1 << i;
+	i++;
+	if (cheri_gettag((void * __capability)frame->tf_elr))
+		regs->tagmask |= (uint64_t)1 << i;
+	i++;
+	if (cheri_gettag((void * __capability)frame->tf_ddc))
+		regs->tagmask |= (uint64_t)1 << i;
```

Comment:
We can do it as a followup, but we should really include the TLS registers in `struct capreg` to make TLS support in debuggers easier.  For legacy aarch64 I still need to add a custom register set to get/set the TLS register, but it would be best to include TLS in the default register set going forward in general on FreeBSD architectures.  (TLS doesn't yet work in gdb on aarch64 due to this whereas it does work on RISC-V, and on x86 it only works "live" but not from cores for the same reason)